### DISCUSSION TITLE
Understanding why my plan cannot be parsed

### DISCUSSION THREAD

**Discussion by alexeh** [AI-Planning/planning-as-a-service#52] [Q&A]
Hi all, first of all thanks for this awesome tool.

I just got into PDDL, and I am running my first plans. I am using PDDL Editor to solve the problems and understand the plan built by different planners. 

However, sometimes I get a "Raw Result" with the full plan concatenated as a single string.

Looking into the code, I believe this is happening here:
https://github.com/AI-Planning/planning-as-a-service/blob/main/server/api/adaptor/planning_editor_adaptor/adaptor.py#L75

And the exception that is catch is being raised here:
https://github.com/AI-Planning/planning-as-a-service/blob/main/server/api/adaptor/planning_editor_adaptor/action_plan_parser/parser.py#L329

The trace is fairly understandable:
```The types found in the problem file must be a subset of the types listed in the domain file\n ```

However I don't see where this is happening, as I don't have any object in the problem "instantiated" as a type that is not defined in my Domain. Is there something basic I am missing?

Plus, I don't understand if I had such a huge mess, how the planner seem to solve the problem

Any idea that what could be happening? Or any advice?

Thanks!

**Comment by haz** [comment]
Can you point us to the PDDL you're using?

**Reply by alexeh** [reply]
Hey @haz sorry for getting back late, I was away for a couple of days. I finally managed to understand what was my error:

In my domain, I had one predicate **without typing**: 

```
 ...predicates
(at ?loc - localization - ?vehicle - transport)
(connected - ?loc ?loc)
 ....more predicates
```

Once I fixed this, I can get the my plan as an "array of actions". Weird stuff is, I haven't properly double-checked it, but when I was reviewing the plan, the algorithm seemed to pick the correct values for the plan?

Could this be arbitrary, or is indeed handled as awell? Would be great to know!

Thanks again for the amazing tool!

**Reply by haz** [reply]
So you mean the actions that appear as full schema on the right? Those aren’t arbitrary. The plan is the actual list, and what you see on the right is the lifted action that’s grounded with the specific objects/constants on each action in your plan.

Glad it finally worked!
